libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -288,6 +297,32 @@ void AP_ICEngine::update(void)
         }
     }
 
+    // update the filtered RPM value
+    if (rpm.get_rpm(rpm_instance-1, rpm_value)) {
+        filtered_rpm_value =  _rpm_filter.apply(rpm_value, AP::scheduler().get_loop_period_s());
+    } else {
+        _rpm_filter.reset();
+    }
+
+    // check against redline RPM
+    if (redline_rpm > 0 && rpm.get_rpm(rpm_instance-1, rpm_value)) {
+        if (filtered_rpm_value > redline_rpm && !(options & uint16_t(Options::DISABLE_REDLINE_WARNING))) {
```

Comment:
the options check should just be for the send texts, otherwise the redline flag is never set so you can't limit but not warn.

Reply:
The other way to think of it is remove the send texts option so the flag always gets set. I think I like that way more. One of my coworkers brought up the case of the only way to save a plane is to give all the power you can get and deal with the engine issues if you can it back on the ground. Fixed.